builtins-default-costs/src/lib.rs

```diff
@@ -14,27 +14,27 @@ use {
 };
 
 /// DEVELOPER: when a builtin is migrated to sbpf, please add its corresponding
-/// migration feature ID to BUILTIN_INSTRUCTION_COSTS, so the builtin's default
-/// cost can be determined properly based on feature status.
+/// migration feature ID to BUILTIN_INSTRUCTION_COSTS, and move it from
+/// NON_MIGRATING_BUILTINS_COSTS to MIGRATING_BUILTINS_COSTS, so the builtin's
+/// default cost can be determined properly based on feature status.
 /// When migration completed, eg the feature gate is enabled everywhere, please
-/// remove that builtin entry from BUILTIN_INSTRUCTION_COSTS.
+/// remove that builtin entry from MIGRATING_BUILTINS_COSTS.
 #[derive(Clone)]
-struct BuiltinCost {
+pub struct BuiltinCost {
     native_cost: u64,
     core_bpf_migration_feature: Option<Pubkey>,
 }
 
-lazy_static! {
-    /// Number of compute units for each built-in programs
-    ///
-    /// DEVELOPER WARNING: This map CANNOT be modified without causing a
-    /// consensus failure because this map is used to calculate the compute
-    /// limit for transactions that don't specify a compute limit themselves as
-    /// of https://github.com/anza-xyz/agave/issues/2212.  It's also used to
-    /// calculate the cost of a transaction which is used in replay to enforce
-    /// block cost limits as of
-    /// https://github.com/solana-labs/solana/issues/29595.
-    static ref BUILTIN_INSTRUCTION_COSTS: AHashMap<Pubkey, BuiltinCost> = [
+/// Number of compute units for each built-in programs
+///
+/// DEVELOPER WARNING: This map CANNOT be modified without causing a
+/// consensus failure because this map is used to calculate the compute
+/// limit for transactions that don't specify a compute limit themselves as
+/// of https://github.com/anza-xyz/agave/issues/2212.  It's also used to
+/// calculate the cost of a transaction which is used in replay to enforce
+/// block cost limits as of
+/// https://github.com/solana-labs/solana/issues/29595.
+pub const MIGRATING_BUILTINS_COSTS: &[(Pubkey, BuiltinCost)] = &[
```

Comment:
minor: can we add a simple test in this module?

```rust
#[test]
fn test_const_builtin_cost_arrays() {
    // sanity check to make sure built-ins are declared in the correct array
    assert!(MIGRATING_BUILTINS_COSTS.iter().all(|(_, c)| c.core_bpf_migration_feature.is_some()));
    assert!(NON_MIGRATING_BUILTINS_COSTS.iter().all(|(_,c)| c.core_bpf_migration_feature.is_none()));
}
```

Reply:
in b6c391bd7010a22c760e5260798d911153a6436e